system-test/pubsub.ts

```diff
@@ -1023,6 +1031,153 @@ describe('pubsub', () => {
     });
   });
 
+  describe('schema', () => {
+    // This should really be handled by a standard method of Array(), imo, but it's not.
+    async function aiToArray(
+      iterator: AsyncIterable<ISchema>,
+      nameFilter?: string
+    ): Promise<ISchema[]> {
+      const result = [] as ISchema[];
+      for await (const i of iterator) {
+        if (!nameFilter || (nameFilter && i.name?.endsWith(nameFilter))) {
+          result.push(i);
+        }
+      }
+
+      return result;
+    }
+
+    const getSchemaDef = async () => {
+      const schemaDef = (
+        await fs.readFile('system-test/fixtures/provinces.avsc')
+      ).toString();
+
+      return schemaDef;
+    };
+
+    const setupTestSchema = async () => {
+      const schemaDef = await getSchemaDef();
+      const schemaId = generateSchemaName();
+      await pubsub.createSchema(schemaId, SchemaTypes.Avro, schemaDef);
+      return schemaId;
+    };
+
+    it('should create a schema', async () => {
+      const schemaId = await setupTestSchema();
+      const schemaList = await aiToArray(pubsub.listSchemas(), schemaId);
+      assert.strictEqual(schemaList.length, 1);
+    });
+
+    it('should delete a schema', async () => {
+      const schemaId = await setupTestSchema();
+
+      // Validate that we created one, because delete() doesn't throw, and we
+      // might end up causing a false negative.
+      const preSchemaList = await aiToArray(pubsub.listSchemas(), schemaId);
+      assert.strictEqual(preSchemaList.length, 1);
+
+      await pubsub.schema(schemaId).delete();
+
+      const postSchemaList = await aiToArray(pubsub.listSchemas(), schemaId);
+      assert.strictEqual(postSchemaList.length, 0);
+    });
+
+    it('should list schemas', async () => {
+      const schemaId = await setupTestSchema();
+
+      const basicList = await aiToArray(
+        pubsub.listSchemas(SchemaViews.Basic),
+        schemaId
+      );
+      assert.strictEqual(basicList.length, 1);
```

Comment:
What about concurrent builds? Will it still suceeds?

Reply:
Good question, though I think it will be okay, because `aiToArray` filters by schema ID. Those are using a unique ID generated per run.